File: app/src/main/java/java/android/notes/core/Note.java
package java.android.notes.core;

public class Note {
    private String date;
    private String headline = "",description = "",body = "";
    private String id;

    public Note() {}
    public Note(String date) {
        this.date = date;
    }

    // get //
    public String getDate() { return date; }

    public String getHeadline() {
        return headline;
    }
    public String getDescription() { return description; }
    public String getBody() { return body; }

    public String getId() { return id; }

    // set //
    public void setDate(String date) {
        this.date = date;
    }

    public void setHeadline(String headline) {
        this.headline = headline;
    }
    public void setDescription(String description) {
        this.description = description;
    }
    public void setBody(String body) { this.body = body; }

    public void setId(String id) { this.id = id; }

    // logic //
    public boolean ready(){
        if(headline.equals("")){ return false; }
        return true;
    }

}

File: app/src/main/java/java/android/notes/core/Note.kt
package java.android.notes.core

class Note {
    // set //
    // get //
    public var date: String = ""
    public var headline = ""
    public var description = ""
    public var body = ""
    public var id: String? = null

    public fun setterHeadLine(text:String){ headline = text }
    public fun setterDescription(text:String){ description = text }
    public fun setterBody(text:String){ body = text }
    public fun setterDate(text:String){ date = text }

    constructor() {}
    constructor(date: String) {
        this.date = date
    }

    // logic //
    fun ready(): Boolean {
        return if (headline == "") {
            false
        } else true
    }

}